src/app/fetcher/fullDomFetcher.js

```diff
@@ -42,9 +38,24 @@ export default async function fetch(url, cssSelectors, { retry } = { retry: 0 })
       throw new FetchDocumentError(`Received HTTP code ${statusCode} when trying to fetch '${url}'`);
     }
 
-    await Promise.all(selectors.map(selector => page.waitForSelector(selector, { timeout: config.get('fetcher.waitForElementsTimeout') })));
+    const mimeType = (response.headers() || [])['content-type'];
+
+    if (mimeType.startsWith('text/')) {
+      content = await response.text();
+    } else {
+      try {
+        await Promise.all(selectors.map(selector => page.waitForSelector(selector, { timeout: config.get('fetcher.waitForElementsTimeout') })));
+        content = await page.content();
+      } catch (e) {
+        // Selectors have not been found in the page
+        // Return the complete snapshot that have been retrieved
+        content = await response.text();
```

Comment:
I think we should only do this when the error is effectively a `waitForElementsTimeout` error like this:

```
if (error instanceof puppeteer.pptr.errors.TimeoutError) { // Expected elements are not present on the web page
    content = await page.content(); // Get the whole page content as content selectors may be wrong
}
```